### PR TITLE
Help key component

### PR DESCRIPTION
**Pull request by Blackbaud-BrandonJones**
Creating PR for review purposes before continuing.

### CODE REVIEW

File: src/app/public/src/modules/help-key/help-key.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        { provide: BBHelpClientService, useValue: mockHelpService }
+      ]
+    })
+      .compileComponents();
+
+    fixture = TestBed.createComponent(HelpKeyComponent);
+    component = fixture.componentInstance;
+  });
+
+  it('should call call the help service\'s setCurrentHelpKey method with it\'s helpKey', () => {
```
Comment (blackbaud-brandonhare) [inline src/app/public/src/modules/help-key/help-key.component.spec.ts]: `should call the help service\'s setCurrentHelpKey method with its helpKey`

File: src/app/public/src/modules/test-button/test-button.component.html [hunk 2]
```diff
@@ -0,0 +1 @@
+<button class="test-button-component" (click)="openWidgetToHelpKey()">Click To Open</button>
```
Comment (blackbaud-brandonhare) [inline src/app/public/src/modules/test-button/test-button.component.html]: This button is just for testing the key component and the service, correct?

**Comment by blackbaud-brandonhare** [comment]
I think this looks good Jones. Seems fairly straightforward to me. I think the component and service will cover the needs of consumers.

File: src/app/public/src/modules/test-button/test-button.component.html (hunk 2 above)
Comment (Blackbaud-BrandonJones) [reply src/app/public/src/modules/test-button/test-button.component.html]: It's mostly for demo purposes yes.  Likely will not be in the actual release as it isn't needed

**Comment by Blackbaud-SteveBrush** [comment]
Have you thought about what should happen if the user travels to another route while the Widget is open? My instinct was that it should close when clicking on a new route...

**Comment by Blackbaud-SteveBrush** [comment]
I'm unsure the extra service layer is necessary, as you're essentially masking the same methods on the Client.

**Comment by Blackbaud-BrandonJones** [comment]
The extra service layer is what gets imported into other components that want to use the open functionality of the widget.   

This is important for things like the `onClick` method stuff that is needed by the skyux components such as `inline-help` or the future modal help button.  

https://github.com/Blackbaud-BrandonJones/bb-help-test-skyux2/blob/master/src/app/test-component/tester.component.ts#L12

Rather than directly using the `BBHelpClient` itself, we use this library to create the one component, and expose a service that can take care of and act as a translation layer for anywhere else that might need it.   This keeps us from exposing some of those methods that are only needed by the widget or builder (load for example) to components that don't need them.

I agree expected behavior would be for the widget to close when navigating between pages,  not sure how we can accomplish that, but It could definitely be seen as a separate issue backlog item for a separate PR since this one is getting hefty, thoughts?

**Comment by Blackbaud-SteveBrush** [comment]
Since this repo represents a component library (the same as SKY UX) I'm leaning toward Model A (no service layer). Any component library that needs to reference the Help Client can do so, and the _client itself_ governs the API, instead of every other component library having to require this library in order to interact with the client.

**Model A**

model-a

**Model B**

model-b

(We can add an issue to close the widget on `NavigationEnd`, in a different PR, if you wish.)

**Comment by Blackbaud-BrandonJones** [comment]
So current behavior of the widget is that it remains open when navigating.   The more I think on this, it almost seems like that is the desired behavior.  If you had a help page open to a specific key,  and wanted to read that information but clicked on another page, you wouldn't want to lose the information you were currently reading, and not be able to easily navigate back to that page you were on.   Thoughts @Blackbaud-SteveBrush ?

**Pull request merged**